src/main/java/com/google/javascript/clutz/ImportRenameMapBuilder.java

```diff
@@ -7,7 +7,19 @@
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * If Clutz is running in incremental mode, closure can't resolve imported symbols, so it gives them
+ * names in the current namespace (ie `module$contents$current$file`), instead of names that
+ * reference the original namespace (ie `module$exports$imported$file`). ImportRenameMapBuilder
+ * builds a map from the local name to the exported name so the original names can be substituted.
+ */
 public class ImportRenameMapBuilder {
+
+  /**
+   * Takes a collection of source files, parses them with the compiler, and walks the ast to find
+   * any `const variable = goog.require()` statements to build a map from the name closure generates
+   * in incremental mode to the exported name of the import.
+   */
```

Comment:
Can buildImportRenameMap be a static method? It's a bit of a code smell to call `new Foo().bar()` without using the instance at all. You could still construct an instance in here that uses a private ctor.

Reply:
It sure can, and would have been my preference, but I was deferring to a preference you had expressed for instance methods over static methods.  Is that just a Java vs. Typescript thing?

Reply:
In TS, I'd make this a top level function, not a static method.